src/main.rs

```diff
@@ -66,7 +63,6 @@ fn main() -> Result<(), aes_gcm_siv::Error> {
         // b_i <-- KA.R
         let b_i: BigInteger256 = rng.gen();
         b_i_array.push(b_i);
-        //let b_i: BigInteger256 = BigInteger256::from(random_u64);
         println!("Random u64 value: {}", b_i);
 
         // m^'_i = KA.msg_2(b_1, m)
```

Comment:
needless borrow in `.pow()`

Reply:
Saw the `&` used in `https://github.com/arkworks-rs/algebra/tree/master/ff#primefield`, that's why. The code works fine also when I don't borrow, yes.